Sources/Bodega/SQLiteStorageEngine.swift

```diff
@@ -29,115 +29,87 @@ import SQLite
 /// One alternative is to make the initializer `throw`, and that's a perfectly reasonable tradeoff.
 /// While that is doable, I believe it's very unlikely the caller will have specific remedies for
 /// specific SQLite errors, so for simplicity I've made the initializer return an optional ``SQLiteStorageEngine``.
-public actor SQLiteStorageEngine: StorageEngine {
+public actor SQLiteStorageEngine {
+    private let tableName: String
+    private let sqliteFileURL: URL
 
-    private let connection: Connection
+    private var dbPointer: OpaquePointer?
 
-    /// A directory on the filesystem where your ``StorageEngine``s data will be stored.
-    private let directory: FileManager.Directory
-
-    /// Initializes a new ``SQLiteStorageEngine`` for persisting `Data` to disk.
-    ///
-    /// - Parameter directory: A directory on the filesystem where your files will be written to.
-    /// `FileManager.Directory` is a type-safe wrapper around URL that provides sensible defaults like
-    ///  `.documents(appendingPath:)`, `.caches(appendingPath:)`, and more.
-    public init?(directory: FileManager.Directory, databaseFilename filename: String = "data") {
-        self.directory = directory
+    /// Initialize a new instance of the ``SQLiteStorageEngine`` for persisting `Data` to disk.
+    /// - Parameters:
+    ///   - directory: The director that will contain the sqlite3 file. `FileManager.Directory` is a type safe wrapper around URL that provides sensible defaults like `.documents(appendingPath:)`, `.caches(appendingPath:)` and more.
+    ///   - filename: The `String` filename to use for the database, this will also be used as the table name in the database.
+    ///   - fileProtection: The `URLFileProtection` used when creating sqlite3 file. **NOTE** Using .complete or .completeUnlessOpen
+    ///   will cause the database to not be able to be read or written to while the app is in the background.
+    init?(
+        directory: FileManager.Directory,
+        databaseFilename filename: String = "data",
+        fileProtection: URLFileProtection = .completeUntilFirstUserAuthentication
+    ) {
+        tableName = filename
 
         do {
-            if !Self.directoryExists(atURL: directory.url) {
-                try Self.createDirectory(url: directory.url)
-            }
+            sqliteFileURL = try Self.createSQLiteFile(in: directory, withFilename: filename, attributes: [.protectionKey: fileProtection])
```

Comment:
@mergesort One thing that I had noticed for both the current `SQLiteStorageEngine` as well as the `DiskStorageEngine` is a lack of the ability to explicitly set the `URLFileProtection` level on the created files on disk.

I've added this here and the default protection level mirrors the system standard, but I was wondering if that was an explicit decision that was made during development? 

From experience I know that setting it to `.completeUnlessOpen` or `.complete` can complicate things when it comes to interacting with files while your app is in the background, but not providing options to set the level is important for security reasons.